Sources/Shared/Models/Album.swift

```diff
@@ -156,6 +161,17 @@ import Foundation
             }
         }
     }
+
+    private func parseInteractions() {
+        guard let dictionary = self.metadata?[Constant.Key.Interactions] as? [String: Any] else { return }
+
+        self.interactions = [String: Any]()
```

Comment:
Can I suggest that you create a local dictionary here, set it up with the interactions you find and then assign it to the property after the for-loop? 

The reason I mention is that I wonder how safe it is to modify this property in this for loop directly in the event that this somehow is called concurrently from multiple threads? It would still not be 100% safe but at tiny bit safer, what do you think?

Reply:
Yes, good call. I'm going to do the same for connections above 👍